pagerduty/user.go

```diff
@@ -212,6 +224,28 @@ func (s *UserService) Create(user *User) (*User, *Response, error) {
 	return v.User, resp, nil
 }
 
+// searchExistingUser searches a user from API given the user email is already taken
+func (s *UserService) searchExistingUser(user *User) (*User, *Response, error) {
+	resp, _, lErr := s.List(&ListUsersOptions{Query: user.Email})
+	if lErr != nil {
+		return nil, nil, fmt.Errorf("user email address is not unique and failed to fetch existing users: %w", lErr)
+	}
+
+	for _, u := range resp.Users {
+		if isSameUser(u, user) {
+			return s.Get(u.ID, &GetUserOptions{})
+		}
+	}
+
+	return nil, nil, fmt.Errorf("user email is already used with different attributes")
```

Comment:
```suggestion
	return nil, nil, fmt.Errorf("[Email has already been taken]")
```
why not leave the original?

Reply:
1. We need to pass in the original error message to the function
2. We want to show here we have some further searches (i.e., first try to create, API returns duplicate, but we compared the existing ones and can't find it, so it's probably set with different fields other than email)